File: prm/lib/models/assignment.dart
class Assignment {
  final int id;
  final String className;
  final String subject;
  final String title;
  final String description;
  final String teacher;
  final int? teacherId;
  final String dueDate;
  final String createdAt;
  final String? fileUrl;

  Assignment({
    required this.id,
    required this.className,
    required this.subject,
    required this.title,
    required this.description,
    required this.teacher,
    this.teacherId,
    required this.dueDate,
    required this.createdAt,
    this.fileUrl,
  });

  factory Assignment.fromJson(Map<String, dynamic> json) {
    return Assignment(
      id: json['id'],
      className: json['className'],
      subject: json['subject'],
      title: json['title'],
      description: json['description'] ?? '',
      teacher: json['teacher'] ?? '',
      teacherId: json['teacherId'] as int?,
      dueDate: json['dueDate'] ?? '',
      createdAt: json['createdAtCustom'] ?? json['createdAt'] ?? '',
      fileUrl: json['fileUrl'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'className': className,
      'subject': subject,
      'title': title,
      'description': description,
      'teacher': teacher,
      'teacherId': teacherId,
      'dueDate': dueDate,
      'createdAt': createdAt,
      'fileUrl': fileUrl,
    };
  }
}

File: be-fschool/src/main/java/com/tinder/befschool/entity/Assignment.java
package com.tinder.befschool.entity;

import jakarta.persistence.*;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Entity
@Table(name = "assignments")
public class Assignment extends Auditable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String className;

    private String subject;

    private String title;

    @Column(columnDefinition = "TEXT")
    private String description;

    private String teacher;

    @Column(name = "teacher_id")
    private Long teacherId;

    private LocalDate dueDate;

    private LocalDateTime createdAtCustom;

    @Column(name = "file_url")
    private String fileUrl;

    // getters and setters

    public String getFileUrl() {
        return fileUrl;
    }

    public void setFileUrl(String fileUrl) {
        this.fileUrl = fileUrl;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getClassName() {
        return className;
    }

    public void setClassName(String className) {
        this.className = className;
    }

    public String getSubject() {
        return subject;
    }

    public void setSubject(String subject) {
        this.subject = subject;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getTeacher() {
        return teacher;
    }

    public void setTeacher(String teacher) {
        this.teacher = teacher;
    }

    public Long getTeacherId() {
        return teacherId;
    }

    public void setTeacherId(Long teacherId) {
        this.teacherId = teacherId;
    }

    public LocalDate getDueDate() {
        return dueDate;
    }

    public void setDueDate(LocalDate dueDate) {
        this.dueDate = dueDate;
    }

    public LocalDateTime getCreatedAtCustom() {
        return createdAtCustom;
    }

    public void setCreatedAtCustom(LocalDateTime createdAtCustom) {
        this.createdAtCustom = createdAtCustom;
    }
}
